consul/consul.go

```diff
@@ -103,35 +104,47 @@ func contains(s []string, str string) string {
 func (r *ConsulAdapter) GetStatus(service *bridge.Service) error {
 
 	registration := new(consulapi.AgentServiceRegistration)
+	registration.ID = service.ID
+	registration.Name = service.Name
+	registration.Port = service.Port
+	registration.Tags = service.Tags
+	registration.Address = service.IP
+	registration.Check = r.buildCheck(service)
+	registration.Meta = service.Attrs
 	var serviceStatus []string
 	var status string
-
-	registration.ID = service.ID
-
-	log.Println("service.ID: ", service.ID)
-	log.Println("service.Name: ", service.Name)
-	log.Println("service.Port: ", service.Port)
-	log.Println("Service.ContainerID: ", service.ContainerID)
-	log.Println("Tags: ", service.Tags)
-	log.Println("service.IP: ", service.IP)
-	log.Println("service.Nodename: ", service.Nodename)
-	log.Println("service.IP: ", string(service.IP))
+	/*
+	log.Println("service.ID: ", service.ID, "service.Name: ",
+		service.Name, "service.Port: ", service.Port, "Service.ContainerID: ",
+		service.ContainerID, "Tags: ", service.Tags, "service.IP: ", service.IP, "service.Nodename: ", service.Nodename)
+	*/
 	ServiceHealthCheck, _, _ := r.client.Health().Checks(service.Name, nil)
 
-	for k, v := range ServiceHealthCheck {
+	for _, v := range ServiceHealthCheck {
 
-		//log.Println("Status:", v.Status)
 		serviceStatus = append(serviceStatus, v.Status)
-
-		log.Println("k:", k, "v:", v)
-		log.Println("serviceStatus:", serviceStatus)
-		//log.Println("###serviceStatus: ", serviceStatus)
-		//log.Println(v)
+		//log.Println("serviceStatus:", serviceStatus)
 		_ = serviceStatus
 	}
 
-	status = contains(serviceStatus, "critical")
-	influxdb.WriteData(service.Name, service.ContainerID, service.Nodename, service.Port, service.IP, status, service.Tags)
+	status = contains(serviceStatus)
+	log.Println("Writing to Influxdb for ", service.Name)
+
+	// update variables for influxdb
+	metrics := &influxdb.Metrics{
+		ServiceName:   service.Name,
+		ContainerID:   service.ContainerID[:12],
+		HostName:      service.Nodename,
+		ServicePort:   service.Port,
+		ServiceIP:     service.IP,
+		ServiceStatus: status,
+		ServiceTags:   service.Tags,
+	}
+
+	influx := influxdb.New()
```

Comment:
Great. Now that we have a client initialiser we can move this to a place where it going to be executed only once for when the code starts-up. Otherwise, we will be acquiring a InfluxDB client every time we need to write a metrics. In contrast, if we only initialise once, we won't be waiting CPU cycles on this anymore. 

I believe we have two ways of doing this:

* We move this line: `influx := influxdb.New()` into the begining where the adapter is created
* We add the package `init()` function that will be initialise, ensured by the language, only once :)